sdk/core/azure-core/src/main/java/com/azure/core/util/paging/ContinuablePagedByIteratorBase.java

```diff
@@ -100,6 +116,36 @@ synchronized void requestPage() {
         this.done = done || (!receivedPages.get() && !isNextAvailable());
     }
 
+    synchronized void requestPageSync() {
+        /*
+         * In the scenario where multiple threads were waiting on synchronization, check that no earlier thread made a
+         * request that would satisfy the current element request. Additionally, check to make sure that any earlier
+         * requests didn't consume the paged responses to completion.
+         */
+        if (isNextAvailable() || done) {
+            return;
+        }
+
+        AtomicBoolean receivedPages = new AtomicBoolean(false);
+
+        syncPageRetriever
+            .getIterable(continuationState.getLastContinuationToken(), defaultPageSize)
+            .forEach(page -> {
+
+                receivedPages.set(true);
+                addPage(page);
+
+                continuationState.setLastContinuationToken(page.getContinuationToken());
+                this.done = continuationState.isDone();
+            });
```

Comment:
Since we are using the sync page retriever, will we get more than one page at a time?